File: nightgrass/src/main/java/com/goodfeel/nightgrass/dto/OrderItemDto.java
package com.goodfeel.nightgrass.dto;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;

@Getter @Setter
@AllArgsConstructor @NoArgsConstructor
public class OrderItemDto {

    private String productName;
    private String imageUrl;
    private Integer quantity;
    private String properties;
    private BigDecimal unitPrice;
}

File: nightgrass/src/main/kotlin/com/goodfeel/nightgrass/dto/OrderItemDto.kt
package com.goodfeel.nightgrass.dto

import com.fasterxml.jackson.core.type.TypeReference
import com.fasterxml.jackson.databind.ObjectMapper
import java.math.BigDecimal

data class OrderItemDto(
    val productName: String,
    val imageUrl: String,
    val quantity: Int,
    val properties: String? = null,
    val unitPrice: BigDecimal,
    var formattedProperties: Map<String, String> = mapOf()
) {
    companion object {
        private val objectMapper = ObjectMapper()
    }

    fun processProperties() {
        formattedProperties = properties?.let {
            objectMapper.readValue(it, object : TypeReference<Map<String, String>>() {})
        } ?: mapOf()
    }
}
